### PR TITLE
fix: only send Explorer role mail if is Explorer

### PR DESCRIPTION
**Pull request by howar31**
# Issue
The Explorer role was assigned while user created.  So we should send role mail if it is explorer. (not blocking it)

# Notice

# Dependency

### CODE REVIEW

File: controllers/user.go [hunk 1]
```diff
@@ -102,10 +102,10 @@ func (mc *MembershipController) SetUser(c *gin.Context) (int, gin.H, error) {
 			"user.Activated.Valid":         user.Activated.Valid,
 			"user.Activated.Time.IsZero()": user.Activated.Time.IsZero(),
 			"roleCheck":                    roleCheck,
-			"sendAssignRoleMail":           !user.Activated.Valid && user.Activated.Time.IsZero() && !roleCheck,
+			"sendAssignRoleMail":           !user.Activated.Valid && user.Activated.Time.IsZero() && roleCheck,
 		}).Info("SetUser Activated Role check")
 
-		if !roleCheck {
+		if roleCheck {
```
Comment (Aylie-Chou) [inline controllers/user.go:108]: Question: would we not need to send explorer email if the `roleCheck` fail ?

Comment (howar31) [reply controllers/user.go:108]: The `roleCheck` currently function as "checking the user has the Explorer role".  So I think if the user was not an Explorer, we should not send Explorer mail to the user.

Comment (Aylie-Chou) [reply controllers/user.go:108]: how about the error case (where `roleCheckErr` != nil) ?

Comment (Aylie-Chou) [reply controllers/user.go:108]: It seems that above flow may miss sending explorer mail for users (the error cases).
I think you might have to check the error handling spec with PM.

File: controllers/user.go [hunk 2]
```diff
@@ -91,23 +91,13 @@ func (mc *MembershipController) SetUser(c *gin.Context) (int, gin.H, error) {
 	if err != nil {
 		return http.StatusInternalServerError, gin.H{"status": "error", "message": "user does not exist"}, err
 	}
-
+	log.WithFields(log.Fields{
+		"user.Activated.Valid":         user.Activated.Valid,
+		"user.Activated.Time.IsZero()": user.Activated.Time.IsZero(),
+		"sendAssignRoleMail":           !user.Activated.Valid && user.Activated.Time.IsZero(),
```
Comment (Aylie-Chou) [inline controllers/user.go:97]: && or || ?

Comment (howar31) [reply controllers/user.go]: fixed

**Review by Aylie-Chou — APPROVED** [review]
LGTM

**Pull request merged**